dlt/destinations/snowflake/snowflake.py

```diff
@@ -68,8 +68,12 @@ def __init__(
         stage_file_path = ""
 
         if bucket_path:
-            # s3 credentials case
-            if bucket_path.startswith("s3://") and staging_credentials and isinstance(staging_credentials, AwsCredentialsWithoutDefaults):
+            # referencing an external s3 stage does not require explicit AWS credentials
+            if bucket_path.startswith("s3://") and stage_name:
+                from_clause = f"FROM '@{stage_name}'"
+                files_clause = f"FILES = ('{urlparse(bucket_path).path.lstrip('/')}')"
```

Comment:
this one was missing - without FILES, we were loading all files in stage.
btw. we had support for external stages but it got removed in code refactor. good catch